crates/pipeline_manager/src/api/pipeline.rs

```diff
@@ -15,23 +15,24 @@ use utoipa::{IntoParams, ToSchema};
 use crate::{
     api::{examples, parse_uuid_param},
     auth::TenantId,
-    db::{storage::Storage, AttachedConnector, DBError, PipelineId, ProgramId, Version},
+    db::{storage::Storage, AttachedConnector, DBError, PipelineId, Version},
 };
 
 use super::{ManagerError, ServerState};
 use uuid::Uuid;
 
 /// Request to create a new pipeline.
 #[derive(Debug, Deserialize, ToSchema)]
+#[serde(deny_unknown_fields)]
 pub(crate) struct NewPipelineRequest {
-    /// Config name.
+    /// Unique pipeline name.
     name: String,
-    /// Config description.
+    /// Pipeline description.
     description: String,
-    /// Program to create config for.
-    program_id: Option<ProgramId>,
-    /// Pipeline configuration parameters.
-    /// These knobs are independent of any connector
+    /// Name of the program to create a pipeline for.
+    program_name: Option<String>,
+    /// Pipeline configuration parameters (e.g. number of workers).
+    /// These knobs are independent of any connector attached to the pipeline.
     config: RuntimeConfig,
     /// Attached connectors.
     connectors: Option<Vec<AttachedConnector>>,
```

Comment:
The attached connector type here should be different from what is the resolved version that is used in PipelineDescr.